src/api/app/models/relationship/add_role.rb

```diff
@@ -0,0 +1,48 @@
+class Relationship::AddRole
```

Comment:
can you unit test it?

Reply:
it's basically tested from the unit tests running add_person and add_group, but I added some rather minimal test.